sympy/ntheory/factor_.py

```diff
@@ -311,6 +311,54 @@ def multiplicity(p, n):
     return m
 
 
+def factmultiplicity(p, n):
+    """
```

Comment:
I recommend:
* change p to b since it refers to the base for which a power is being sought
* change prime to p
* move the intitialization of m to after the input sanitization
* use `as_int` on the input
* leverage existing functions to compute this (and to avoid the extra multiplicity calc at the end). Using the current notation:
```python
factors = factorint(p)
min_e = min(
    # the exponent of prime in n! = (n - sum(digits of n in base `prime`)/(prime - 1)
    # note: subtract prime since digits includes prime as first element
    (n - (sum(digits(n, prime)) - prime))//(prime - 1)
    for prime in factors)
return min_e//min(factors.values())
```
Otherwise, more descriptive variables names than `y` and `ans` and use of `factorint` instead of `primefactors` would be ok, too.

Although I don't like long names I would prefer `multiplicity_in_factorial` for this function. And although that will be shorter than sending an unevaluated factorial to `multiplicity` I still think a check for an unevaluated factorial should be made in `multiplicity`.
```
>>> multiplicity_in_factorial(42, 10**10)
1666666660
>>> multiplicity(42, factorial(10**10, evaluate=False))
1666666660
```

Reply:
Sure will make the changes and commit soon

Reply:
- move the intitialization of m to after the input sanitization
- use as_int on the input

Can you elaborate this...

Reply:
> Although I don't like long names I would prefer `multiplicity_in_factorial` for this function. And although that will be shorter than sending an unevaluated factorial to `multiplicity` I still think a check for an unevaluated factorial should be made in `multiplicity`.
> 
> ```
> >>> multiplicity_in_factorial(42, 10**10)
> 1666666660
> >>> multiplicity(42, factorial(10**10, evaluate=False))
> 1666666660
> ```

Where should i add the check, in multiplicity, in doctests?